File: src/main/java/org/gradle/plugins/site/generator/SiteGenerator.java
package org.gradle.plugins.site.generator;

import org.gradle.plugins.site.data.CustomData;
import org.gradle.plugins.site.data.ProjectDescriptor;

/**
 * The site generator used to produce web page.
 * <p>
 * The default implementation of this interface is {@link org.gradle.plugins.site.generator.FreemarkerSiteGenerator}.
 */
public interface SiteGenerator {

    void generate(ProjectDescriptor projectDescriptor, CustomData customData);
}

File: src/main/kotlin/org/gradle/plugins/site/generator/SiteGenerator.kt
package org.gradle.plugins.site.generator

import org.gradle.plugins.site.data.ProjectDescriptor
import org.gradle.plugins.site.data.ProjectLinksDescriptor

/**
 * The site generator used to produce web page.
 *
 * The default implementation of this interface is [org.gradle.plugins.site.generator.FreemarkerSiteGenerator].
 */
interface SiteGenerator {
    fun generate(projectDescriptor: ProjectDescriptor, projectLinksDescriptor: ProjectLinksDescriptor)
}
